pyasic/miners/backends/luxminer.py

```diff
@@ -146,6 +147,32 @@ async def reboot(self) -> bool:
     async def get_config(self) -> MinerConfig:
         return self.config
 
+    async def upgrade_firmware(self) -> dict:
+        """
+        Upgrade the firmware on a LuxOS miner by calling the 'updaterun' API command.
+
+        Returns: Result of the upgrade process.
+
+        """
+        command = "updaterun"
+
+        try:
+            response = await self.rpc.send_command(command=command)
+
+            if not response:
+                raise ValueError("No response received during the firmware upgrade process.")
+
+            if response.get("status") == "success":
```

Comment:
Don't need this check, it's already checked in RPC.  If the command is not successful, it will raise `APIError` as below.

Reply:
Additionally, even if this check was needed, it isnt correct.  See https://docs.luxor.tech/firmware/api/luxminer/updaterun#examples